protocol.cpp

```diff
@@ -203,13 +203,53 @@ int redis_protocol::authenticate(const char *credentials)
     int size = 0;
     assert(credentials != NULL);
 
-    size = evbuffer_add_printf(m_write_buf,
-        "*2\r\n"
-        "$4\r\n"
-        "AUTH\r\n"
-        "$%u\r\n"
-        "%s\r\n",
-        (unsigned int)strlen(credentials), credentials);
+    /* Credentials may be one of:
+     * <PASSWORD>           For Redis <6.0 simple AUTH commands.
+     * <USER>:<PASSWORD>    For Redis 6.0+ AUTH with both username and password.
+     *
+     * A :<PASSWORD> will be handled as a special case of quoting a password that
```

Comment:
do we need to think of a case where the USER may contain colon?
what about ask for --user and --password in the command line?

Reply:
Generally we already support the `<USER>:<PASSWORD>` notation on `--authenticate` when supporting binary memcache, so it made more sense to apply that to Redis as well rather than introduce new arguments.

I think not supporting a username with a colon is reasonable. Old-style passwords with a colon are supported (by prepending a colon) so backwards compatibility is maintained.